File: 1012 Complement of Base 10 Integer.java
class Solution {
    public int bitwiseComplement(int N) {
        int X = 1;
        while (N > X) X = X * 2 + 1;
        return X - N;
    }
}

/* Solution from msalto
class Solution {
    public int bitwiseComplement(int N) {
        if (N == 0) {
            return 1;
        }
        int mask = (1 << 30);
        while ((mask & N) == 0) {
            mask >>= 1;
        }
        return (mask - 1) & (~N);
    }
}
*/

File: 1012 Complement of Base 10 Integer.py
class Solution:
    def bitwiseComplement(self, N: int) -> int:
        bf = list(str(bin(N)[2:]))
        res = ['0' if x == '1' else '1' for x in bf]
        return int(''.join(res), 2)

''' Solution from lee215
def bitwiseComplement(self, N):
    X = 1
    while N > X: X = X * 2 + 1
    return X - N

def bitwiseComplement(self, N):
    X = 1
    while N > X: X = X * 2 + 1;
    return N ^ X
'''